### ISSUE TITLE
Default id column has become mandatory during new row when hidden on detail turned on but hidden on browse is turned off

### ISSUE THREAD

**Issue by ithiru** [directus/app#2447]
Version - v8.4.0
Error Message - id: This value should not be blank.

Steps

- Created new collection - turn on the Sort toggle, Modified On and Modified By
- Edit the Default primary key column 'id' by selecting it
- Go to schema Tab, Turn off 'Hidden on Browse'
- Click on 'Next'
- Click on 'Save'
- Go to the newly created collection to add new Row
- Click  on '+' at top right
- You will see only status with default 'Draft' selected - Leave it as is
- Click on 'Tick' button at top right.
- Displays Error Message 'id: This value should not be blank.'

**Comment by philleepflorence** [comment]
NOTE: It seems this "error" is triggered when the ID Field in the Collection is not set to **Auto Increment**.

**Comment by rijkvanzanten** [comment]
Ah in that case it makes perfect sense that the error is returned. If you don't have auto increment enabled, and you don't pass a primary key manually, the API will error out as a primary is always required.

**Comment by philleepflorence** [comment]
By default, Auto Increment should be added to the Primary Int Key, no?
It seems the only change made above was hiding on browse.

**Comment by rijkvanzanten** [comment]
Correct, auto-increment is enabled by default. @ithiru Could you confirm if auto increment is turned on on your end?

**Comment by ithiru** [comment]
> 
> 
> Correct, auto-increment is enabled by default. @ithiru Could you confirm if auto increment is turned on on your end?

is there a way we can turn on the auto increment through UI? I don't see an option to do that. I just did the steps mentioned above.

**Comment by benhaynes** [comment]
@ithiru — if your created the collection through Directus it would have automatically created an `id` column/field with auto increment on. Something seems amiss here. Could you check the database directly to see if that table's primary key has auto increment?

**Comment by benhaynes** [comment]
Also, as far as I know, there's no way to change the auto-increment from the Directus App... so I don't think this happened on our side. Was any part of this table created/edited directly within the database?

**Comment by ithiru** [comment]
@benhaynes I just simulated this @ demo.directus.io by following the steps I provided above.  https://demo.directus.io/admin/#/thumper/collections/testprimaryautoincrement/+ please check..

**Comment by benhaynes** [comment]
Thanks! I have confirmed your issue... very weird indeed!

**Comment by ithiru** [comment]
Thanks for making it as critical, it is indeed..
